alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultNotificationManager.java

```diff
@@ -175,25 +179,35 @@ public PageRequest getPageRequestForNotifications(final Integer pageNumber, fina
         return PageRequest.of(page, size, new Sort(sortingOrder, sortingField));
     }
 
-    private void deleteAuditEntries(final Long notificationId) {
-        final List<AuditNotificationRelation> foundRelations = auditNotificationRepository.findByNotificationId(notificationId);
-        final Function<AuditNotificationRelation, Long> transform = AuditNotificationRelation::getAuditEntryId;
-        final List<Long> auditIdList = foundRelations.stream().map(transform).collect(Collectors.toList());
-        auditNotificationRepository.deleteAll(foundRelations);
-        final List<AuditEntryEntity> auditEntryList = auditEntryRepository.findAllById(auditIdList);
-        auditEntryRepository.deleteAll(auditEntryList);
+    private void deleteAuditEntries(Long notificationId) {
+        List<AuditNotificationRelation> foundRelations;
+        try {
```

Comment:
Here's where the change really is.  try catch block for the old code that was failing.  Attempt to delete from the audit relation by notification id.  Otherwise if it fails log it and move on to deleting the notifications.

Reply:
Does this have to do with foreign keys?

Reply:
Yes I believe it has to do with the foreign key constraints between the audit_notification_relation, audit_entry, and raw_notification_content tables.  Currently the customer is in a state

I can't produce the problem in 5.1.0 and 5.3.0. Because the database is in this odd state in 5.0.0 this change will be able to allow notifications to be purged to clean up the database. Alert may have been missing some delete on cascade constraints.